src/main/scala/org/broadinstitute/dsde/agora/server/webservice/AddHandler.scala

```diff
@@ -53,4 +53,69 @@ class AddHandler(authorizationProvider: AuthorizationProvider) extends Actor {
     val method = agoraBusiness.insert(agoraEntity.copy(createDate = Option(new DateTime())), username)
     context.parent ! RequestComplete(spray.http.StatusCodes.Created.intValue, method)
   }
+
+}
+
+case class MethodImportResolver(username: String, business: AgoraBusiness, authorizationProvider: AuthorizationProvider) {
+
+  def importResolver(importUri: String): WdlSource = {
+    ImportResolverHelper.validateUri(importUri)
+    val method = ImportResolverHelper.resolve(importUri, business, username)
+    val authorizedMethod = authorizationProvider.filterByReadPermissions(method, username)
+    authorizedMethod match {
+      case x: Some[AgoraEntity] => x.get.payload.get
+      case None => throw new SyntaxError(s"Can't resolve import: $importUri")
+    }
+  }
+}
+
+object ImportResolverHelper {
+
+  private val methodScheme = "methods://"
+
+  def resolve(uri: String, business: AgoraBusiness, username: String): Option[AgoraEntity] = {
+    val withoutScheme = removeScheme(uri)
```

Comment:
Would it make sense to create a case class that can take a uri and do all of the parsing internally so that we can use it elsewhere if we need to?